3.10/arangosearch-nested-search.md

```diff
@@ -0,0 +1,218 @@
+to query the nested objects. The default `identity` Analyzer is used for the
+fields because none is specified explicitly.
+
+## Defining how often the conditions need to be true
+
+You can optionally specify a quantifier to define how often the conditions need
+to be true for the entire array. The following query matches documents that have
+one or two nested objects with a height greater than 40:
+
+```aql
+FOR doc IN viewName
+  SEARCH doc.dimensions[? 1..2 FILTER CURRENT.type == "height" AND CURRENT.value > 40]
+  RETURN doc
+```
+
+If you leave out the quantifier, it defaults to `ANY`. The conditions need to be
+fulfilled by at least one sub-object, but more than one sub-object may meet the
+conditions. With a quantity of `1`, it would need to be one match exactly.
+Similarly, ranges require an exact match between the minimum and maximum number,
+including the specified boundaries. To require two or more sub-objects to
+fulfill the conditions, you can use `AT LEAST (2)`, and so on.
+
+{% hint 'info' %}
+The `ALL` quantifier of the Question mark operator is not supported in View queries.
+
+Using the question mark operator without quantifier and filter conditions (`[?]`)
+cannot utilize View indexes.
+{% endhint %}
+
+## Searching deeply nested data
+
+You can index and search for multiple levels of objects in arrays.
+Consider the following document:
+
+```json
+{
+  "dimensions": [
+    {
+      "part": "frame",
+      "measurements": [
+        { "type": "height", "value": 47 },
+        { "type": "width", "value": 72 }
+      ],
+      "comments": "Slightly damaged at the bottom right corner."
+    },
+    {
+      "part": "canvas",
+      "measurements": [
+        { "type": "height", "value": 35 },
+        { "type": "width", "value": 60 }
+      ]
+    }
+  ]
+}
+```
+
+To index the array of dimension objects and the nested array of measurement
+objects, you can use a View definition like the following:
+
+```json
+{
+  "links": {
+    "<collection>": {
+      "fields": {
+        "dimensions": {
+          "nested": {
+            "measurements": {
+              "nested": {
+                "type": {},
+                "value": {}
+              }
+            },
+            "part": {},
+            "comments": {
+              "analyzers": [
+                "text_en"
+              ]
+            }
+          }
+        }
+      }
+    }
+  }
+}
+```
+
+The default `identity` Analyzer is used for the `type`, `value`, and `part`
+attributes, and the built-in `text_en` is used for the `comments`.
+
+A possible query is to search for frames with damaged corners that are not wider
+than 80, using a question mark operator to check the `part` and `comments`, and
+a nested question mark operator to check the `type` and `value`:
+
+```aql
+FOR doc IN viewName
+  SEARCH doc.dimensions[? FILTER CURRENT.part == "frame" AND
+         ANALYZER(TOKENS("corner damage", "text_en") ALL == CURRENT.comments, "text_en") AND
```

Comment:
As for inverted indexes and `search-alias` views we discourage using `ANALYZER` function since actual field value is inferred from a field properties automatically, I suggest we either leave a comment or refactor an example.

Reply:
Nested search using inverted indexes needs to be covered in a separate PR, on a different page, but the `search-alias` case needs to be addressed here, indeed. Should there be separate sections about the different View types but on the same page? Or only remarks about using the ANALYZER() function in combination with search-alias Views? That would be on every example page, however...

Reply:
In the PR for the inverted index, we will remove ANALYZER() if possible and otherwise have two examples, for a Search Alias View and a traditional one (many functions accept an optional analyzer argument, e.g. PHRASE)